server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DomainMappingsRoutes.java

```diff
@@ -114,28 +126,56 @@ public HaltException removeDomainMapping(Request request, Response response) {
             @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
                     message = "Internal server error - Something went bad on the server side.")
     })
-    public Map<String, List<String>> get(Request request, Response response) throws RecipientRewriteTableException {
+    public Map<String, List<String>> getAllMappings(Request request, Response response) throws RecipientRewriteTableException {
         return recipientRewriteTable.getAllMappings()
                 .entrySet()
                 .stream()
-                .collect(toImmutableMap(e -> e.getKey().getFixedDomain(),
-                        e -> e.getValue()
-                                .select(Mapping.Type.Domain)
-                                .asStream()
-                                .map(Mapping::asString)
-                                .map(Mapping.Type.Domain::withoutPrefix)
-                                .collect(toImmutableList())
+                .filter(mappingsEntry -> !mappingsEntry.getValue().isEmpty())
+                .filter(mappingsEntry -> mappingsEntry.getValue().contains(Mapping.Type.Domain))
+                .collect(Guavate.toImmutableMap(
+                        mappingsEntry -> mappingsEntry.getKey().getFixedDomain(),
+                        mappingsEntry -> toDomainList(mappingsEntry.getValue())
                 ));
     }
 
-    private void doMapping(final Request request, final ThrowingBiConsumer<MappingSource, Domain> op) {
-        MappingSource fromDomain = createDomainOrThrow()
+    @GET
+    @Path(SPECIFIC_MAPPING_PATH)
+    @ApiOperation(value = "Lists mappings for specific domain.")
+    @ApiImplicitParams({
+            @ApiImplicitParam(required = true, dataType = "string", name = FROM_DOMAIN, paramType = "path")
+    })
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.OK_200, message = "Domain mappings.", responseContainer = "List"),
+            @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "Not existing mappings."),
+            @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+                    message = "Internal server error - Something went bad on the server side.")
+    })
+    public List<String> getMapping(Request request, Response response) throws RecipientRewriteTableException {
+        final MappingSource mappingSource = mappingSourceFrom(request);
```

Comment:
We do not use final variable :-)

Reply:
It probably deserves some more rationals : 
* we don't allow variable reallocation in our code because it makes reading harder
* in java 8, you don't really need variable to be final to use them in lambda (what previously required an anonymous class + final variable)
* finally, final becomes useless because of the two previous point

Reply:
I should change the IDE preferences on that 👍